File: app/sopapp/lib/models/project.dart
class Project {
  final int id;
  final String title;
  final String researchArea;
  final String status;
  final String description;
  final DateTime startDate;
  final DateTime endDate;
  final String facultyId;

  Project(this.id, this.description, this.startDate, this.endDate, this.facultyId, 
      {required this.title, required this.researchArea, required this.status});
}

File: server/models/project.js
// models/Project.js
const { DataTypes } = require('sequelize');
const sequelize = require('server/sequelize.js');
const User = require('server/models/user.js');

const Project = sequelize.define('Project', {
  id: {
    type: DataTypes.UUID,
    primaryKey: true,
    defaultValue: DataTypes.UUIDV4,
  },
  GPSRN : DataTypes.STRING,
  title: DataTypes.STRING,
  description: DataTypes.STRING,
  researchArea: DataTypes.STRING,
  comments : DataTypes.STRING,
  status : DataTypes.ENUM('Inactive','Active'),
  startDate: DataTypes.DATE,
  endDate: DataTypes.DATE,
  comments : DataTypes.STRING
});

Project.belongsTo(User, { as: 'faculty', foreignKey: 'facultyId' });

module.exports = Project;
